design/open/spark-outputcommitter/perf.md

```diff
@@ -0,0 +1,173 @@
+  "func": "pkg/api.(*Controller).handleAPIError",
+  "host": "XXXX.us-east-1.lakefscloud.ninja",
+  "level": "error",
+  "log_audit": "API",
+  "method": "GET",
+  "msg": "API call returned status internal server error",
+  "operation_id": "DiffRefs",
+  "path": "/api/v1/repositories/test-1/refs/lakeFS-OC--1l6fxphkzwaascr80nbp2tiour6kztr3asyov0k8stzt6u1wdn-lakefs---test-1-test-clean-2-amazon-reviews-repartition-clean2-csv-attempt_202211071322498107897909273126199_0003_m_003334_3843/diff/lakeFS-OC--1l6fxphkzwaascr80nbp2tiour6kztr3asyov0k8stzt6u1wdn-lakefs---test-1-test-clean-2-amazon-reviews-repartition-clean2-csv?amount=1&type=two_dot&diff_type=two_dot",
+  "request_id": "985c1378-32b5-439f-b4cf-2334c5126d59",
+  "service": "api_gateway",
+  "service_name": "rest_api",
+  "time": "2022-11-07T13:56:17Z"
+}
+{
+  "error": "open range c85d0ca8c208e200527f20fd78c58866e92e83e9f676da528b64e868fb13a091: open sstable reader s3://yoni-test3/data/XXXX c85d0ca8c208e200527f20fd78c58866e92e83e9f676da528b64e868fb13a091: pebble/table: invalid table (bad magic number)",
+  "file": "build/pkg/api/controller.go:1745",
+  "func": "pkg/api.(*Controller).handleAPIError",
+  "host": "XXXX.us-east-1.lakefscloud.ninja",
+  "level": "error",
+  "log_audit": "API",
+  "method": "GET",
+  "msg": "API call returned status internal server error",
+  "operation_id": "DiffRefs",
+  "path": "/api/v1/repositories/test-1/refs/lakeFS-OC--1l6fxphkzwaascr80nbp2tiour6kztr3asyov0k8stzt6u1wdn-lakefs---test-1-test-clean-2-amazon-reviews-repartition-clean2-csv-attempt_202211071232381584428887595320545_0007_m_002678_6709/diff/lakeFS-OC--1l6fxphkzwaascr80nbp2tiour6kztr3asyov0k8stzt6u1wdn-lakefs---test-1-test-clean-2-amazon-reviews-repartition-clean2-csv?amount=1&type=two_dot&diff_type=two_dot",
+  "request_id": "9b7b65ff-6278-4fc4-b6e3-b3377014f862",
+  "service": "api_gateway",
+  "service_name": "rest_api",
+  "time": "2022-11-07T12:55:38Z"
+}
+```
+
+RPS is more important for us here, so average latency is the key.  However
+it is hard to estimate total merge time from merge operation latencies,
+because most of the wasted work on a merge is work that occurs in parallel
+with other merges.
+
+## Suggestions
+
+### Find "silver bullet" to boost merge performance on lakeFS Cloud
+
+### Measure merge performance
+
+Effects of retries are unclear; if we are using poor parameters we will be
+going much too slowly.  Currently we just use the default backoff
+parameters, which might not fit our use-case.  Measure and find good
+parameters.
+
+### Boost merge performance
+
+Consider speeding up implementation of the algorithm suggested in [#4057].
+
+### More magic-like LakeFSOutputCommitter
+
+Use unlinked objects on S3 as task temporaries.  Requires modifying
+LakeFSFileSystem or building something on top of it.  (Design forthcoming).
+
+#### Alternative: move objects from task branch to job branch
+
+### Don't do multiple merges as part of task but only at end of jobs
```

Comment:
do lines 145 and 155 describe the same experiment?

Reply:
This is the start, but not concurrent.